mkdocs setup (deploy, dev, user-guide)

Shrinks99 opened the pull request:
- Includes docs contribution instructions in the main readme
- Sets up mkdocs
  - There are probably some remaining settings and flags we want to use but that can be sorted out as time goes on?
- Moves front end dev readme & deployment readme to /docs
- ⚠️ Does not move the Ansible playbook readme because frankly I'm unsure what this should be titled over in /docs & how it's related to deployment.  A decision for somebody else to make perhaps... (@ikreymer)

### Next steps

- Set up versioning with [Mike](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/)
- Turn on readthedocs hosting

Closes #336

Shrinks99 committed:
Initial docs move

Moves front end dev info & deployment info to mkdocs

Shrinks99 committed:
Update index.md

Shrinks99 committed:
Adds instructions for building docs

SuaYoo commented on README.md:
```diff
@@ -14,21 +14,27 @@ The system is designed to run in both Kubernetes and Docker Swarm, as well as lo
+## Documentation
 
-Browsertrix Cloud is made available under the AGPLv3 License.
+User and developer docs are located in /docs. We use [MkDocs](https://squidfunk.github.io/mkdocs-material/) with the [Material theme](https://squidfunk.github.io/mkdocs-material/).
```
Nitpick: Is it important to mention the Material theme here, since the README is so sparse?

Shrinks99 replied:
Ehh, could take it or leave it.  Material for MkDocs has a bunch of flags and features that it ships with and that site is a useful reference, but the only reason this link is here is to save a Google search really.  Also I feel like people could sus out that the docs are located in `/docs` because that's pretty standard so if we're removing the link to the theme we may as well remove the rest of the line too? :P

SuaYoo commented on docs/developer-documentation/front-end-development.md:
```diff
@@ -1,4 +1,4 @@
-# Browsertrix Cloud frontend
+# Front-End Development
```
```suggestion
# Frontend Development
```

Nitpick: I believe it's more common/correct when using as a noun (e.g. "I'm working on the Browsertrix frontend" v.s. "I'm a front-end developer working on Browsertrix")

SuaYoo reviewed (commented):
Nice! Super easy to set up, and clear where the docs are. Just some minor comments.

Shrinks99 committed:
Update docs/developer-documentation/front-end-development.md

Co-authored-by: sua yoo <sua@suayoo.com>

Shrinks99 committed:
Add note about checking your directory

I didn't do this and things didn't work properly so lets make it explicit!

ikreymer committed:
docs work:
- add new deployment docs, local and prod
- set up three sections: deployment, dev and user guide
- remove old deployment docs

ikreymer committed:
add mkdocs gh-pages publish
update README

ikreymer committed:
tweaks to docs

ikreymer committed:
additional tweaks to docs

ikreymer committed:
fix links

ikreymer committed:
add dev/index.md

ikreymer commented:
This now includes the initial docs setup with three sections, including new deployment guide, dev docs, and user-guide placeholder.
Also includes CI build of docs on github pages.
